.github/workflows/deploy-cloud.yaml

```diff
@@ -34,32 +40,51 @@ jobs:
-        uses: aws-actions/configure-aws-credentials@v1
+        if: env.REGION == 'us'
+        uses: aws-actions/configure-aws-credentials@v4
         with:
-          aws-access-key-id: ${{ secrets.AWS_DASHBOARD_PROD_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_DASHBOARD_PROD_SECRET_ACCESS_KEY }}
+          aws-access-key-id: ${{ secrets.AWS_CLOUD_US_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_US_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
+
       - name: Deploy
+        if: env.REGION == 'eu'
         run: |
-          if [[ "$REGION" == "us" ]]; then
-            BUCKET=${{ secrets.AWS_PROD_US_DEPLOYMENT_BUCKET }}
-            CF_ID=${{ secrets.AWS_PROD_US_CF_DIST_ID }}
-          elif [[ "$REGION" == "eu" ]]; then
-            BUCKET=${{ secrets.AWS_PROD_EU_DEPLOYMENT_BUCKET }}
-            CF_ID=${{ secrets.AWS_PROD_EU_CF_DIST_ID }}
-            CF_2_ID=${{ secrets.AWS_PROD_EU_CF_2_DIST_ID }}
-          else
-            echo "Unknown region provided"
-            exit 1
-          fi
+          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
+          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
+
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
+          done
+          env:
+            DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_EU_DEPLOYMENT_BUCKET }}
+            CDN_DISTRIBUTIONS: ${{ secrets.AWS_CLOUD_EU_CDN_DISTRIBUTIONS }}
+
+            - name: Deploy
+            if: env.REGION == 'us'
+            run: |
+              aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
+              aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
+
+              IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+              for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+                for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
+              done
+              env:
+                DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_US_DEPLOYMENT_BUCKET }}
+                CDN_DISTRIBUTIONS: ${{ secrets.AWS_CLOUD_US_CDN_DISTRIBUTIONS }}
 
-          aws s3 sync build/dashboard s3://${BUCKET}/${ENVIRONMENT}/static/
-          aws s3 cp build/dashboard/index.html s3://${BUCKET}/${ENVIRONMENT}/
-          for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${CF_ID} --paths "/dashboard*" && break || sleep 5; done
-          if [[ -n "$CF_2_ID" ]]; then
-            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${CF_2_ID} --paths "/dashboard*" && break || sleep 5; done
```

Comment:
that allow us to invalidate all 4 distributions we currently have in EU, and if another will be introduced, we can just update secret and it will work just fine